python/cudf/cudf/_lib/cudf.pyx

```diff
@@ -396,93 +329,82 @@ cdef gdf_column_to_column(gdf_column* c_col, int_col_name=False):
         A flag indicating the string column name should be cast
         to an integer after decoding (default: False).
     """
-    from cudf.core.column import Column
+    from cudf.core.buffer import Buffer
+    from cudf.core.column import build_column, as_column
+    from cudf.utils.utils import mask_bitsize, calc_chunk_size
+
     name = None
-    ncount = c_col.null_count
     if c_col.col_name is not NULL:
         name = c_col.col_name.decode()
         if int_col_name:
             name = int(name)
-    data, mask = gdf_column_to_column_mem(c_col)
-    return Column.from_mem_views(data, mask, ncount, name)
+    
+    gdf_dtype = c_col.dtype
+    data_ptr = int(<uintptr_t>c_col.data)
 
-
-cdef gdf_column_to_column_mem(gdf_column* input_col):
-    gdf_dtype = input_col.dtype
-    data_ptr = int(<uintptr_t>input_col.data)
     if gdf_dtype == GDF_STRING:
         data = nvstrings.bind_cpointer(data_ptr)
+        result = as_column(data, name=name)
     elif gdf_dtype == GDF_STRING_CATEGORY:
         # Need to do this just to make sure it's freed properly
         garbage = rmm.device_array_from_ptr(
             data_ptr,
-            nelem=input_col.size,
+            nelem=c_col.size,
             dtype='int32',
             finalizer=rmm._make_finalizer(data_ptr, 0)
```

Comment:
would be nice to just call rmmFree here instead of using Numba to effectively do that.

Reply:
Done. We call `c_free` now